In [2]:
pwd

'C:\\Users\\91918\\rna seq'

#import os
os.chdir(r"C:\Users\91918")  # Set working directory
print(os.listdir())  # Check available files

In [12]:
import pandas as pd
df1 = pd.read_excel("resLFC_Y_B6J_4w_AL_vs_NR.xlsx")

In [13]:
df1

,Unnamed: 0,baseMean,log2FoldChange,lfcSE,pvalue,padj,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,gene,log2FoldChange.1
0,Xkr4,1.375127,-0.000082,0.043642,0.935704,NaN,NaN,NaN,NaN,NaN,Xkr4,-0.000082
1,Gm7341,1.311742,0.001165,0.043656,0.431984,NaN,NaN,NaN,NaN,NaN,Gm7341,0.001165
2,Sox17,31.231318,-0.002580,0.043276,0.686624,0.938279,NaN,NaN,NaN,NaN,Sox17,-0.002580
3,Gm7357,5.921844,0.000134,0.043562,0.997165,0.999089,NaN,NaN,NaN,NaN,Gm7357,0.000134
4,Gm6085,50.962861,-0.000125,0.042967,0.980746,0.995715,NaN,NaN,NaN,NaN,Gm6085,-0.000125
...,...,...,...,...,...,...,...,...,...,...,...,...
22873,AC125149.1,5.412648,-0.002955,0.043672,0.385487,0.895400,NaN,NaN,NaN,NaN,AC125149.1,-0.002955
22874,AC168977.2,2.975387,0.001831,0.043649,0.476003,0.905988,NaN,NaN,NaN,NaN,AC168977.2,0.001831
22875,AC168977.1,1.426839,-0.002510,0.043744,0.085025,NaN,NaN,NaN,NaN,NaN,AC168977.1,-0.002510
22876,AC149090.1,584.916210,0.016432,0.046439,0.167588,0.884483,NaN,NaN,NaN,NaN,AC149090.1,0.016432


In [14]:
df1.head(3)

,Unnamed: 0,baseMean,log2FoldChange,lfcSE,pvalue,padj,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,gene,log2FoldChange.1
0,Xkr4,1.375127,-0.000082,0.043642,0.935704,NaN,NaN,NaN,NaN,NaN,Xkr4,-0.000082
1,Gm7341,1.311742,0.001165,0.043656,0.431984,NaN,NaN,NaN,NaN,NaN,Gm7341,0.001165
2,Sox17,31.231318,-0.002580,0.043276,0.686624,0.938279,NaN,NaN,NaN,NaN,Sox17,-0.002580


In [15]:
df1.columns

Index(['Unnamed: 0', 'baseMean', 'log2FoldChange', 'lfcSE', 'pvalue', 'padj',
       'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'gene',
       'log2FoldChange.1'],
      dtype='object')

In [16]:
df1.tail(3)

,Unnamed: 0,baseMean,log2FoldChange,lfcSE,pvalue,padj,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,gene,log2FoldChange.1
22875,AC168977.1,1.426839,-0.002510,0.043744,0.085025,NaN,NaN,NaN,NaN,NaN,AC168977.1,-0.002510
22876,AC149090.1,584.916210,0.016432,0.046439,0.167588,0.884483,NaN,NaN,NaN,NaN,AC149090.1,0.016432
22877,CAAA01147332.1,469.139410,0.003584,0.042323,0.750999,0.952547,NaN,NaN,NaN,NaN,CAAA01147332.1,0.003584


In [21]:
unique_genes=df1["gene"].unique()
print(unique_genes)

['Xkr4' 'Gm7341' 'Sox17' ... 'AC168977.1' 'AC149090.1' 'CAAA01147332.1']


In [23]:
df1.isna

<bound method DataFrame.isna of            Unnamed: 0    baseMean  log2FoldChange     lfcSE    pvalue  \
0                Xkr4    1.375127       -0.000082  0.043642  0.935704   
1              Gm7341    1.311742        0.001165  0.043656  0.431984   
2               Sox17   31.231318       -0.002580  0.043276  0.686624   
3              Gm7357    5.921844        0.000134  0.043562  0.997165   
4              Gm6085   50.962861       -0.000125  0.042967  0.980746   
...               ...         ...             ...       ...       ...   
22873      AC125149.1    5.412648       -0.002955  0.043672  0.385487   
22874      AC168977.2    2.975387        0.001831  0.043649  0.476003   
22875      AC168977.1    1.426839       -0.002510  0.043744  0.085025   
22876      AC149090.1  584.916210        0.016432  0.046439  0.167588   
22877  CAAA01147332.1  469.139410        0.003584  0.042323  0.750999   

           padj  Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  \
0           NaN         

In [25]:
num_unique_genes=df1["gene"].nunique()
print(f"number of unique genes: {num_unique_genes}")

number of unique genes: 22878
